build.js

```diff
@@ -1,34 +1,125 @@
+const VSCODE_VERSION = version.split("-")[0];
+
+function error(msg) {
+  console.info("\x1b[31merror %s\x1b[0m", msg)
+}
+function ok(msg) {
+  console.info("\x1b[32m%s\x1b[0m", msg)
+}
+function note(msg) {
+  console.info("\x1b[90m%s\x1b[0m", msg)
+}
+function exec(cmd, opts) {
+  console.info("\x1b[36m%s\x1b[0m", cmd)
+  return child_process.execSync(cmd, opts);
+}
+
+const requiredTools = ["node", "yarn", "git", "python"];
+note(`required tools ${JSON.stringify(requiredTools)}`)
+for (const tool of requiredTools) {
+  try {
+    child_process.execSync(`${tool} --version`, { stdio: "ignore" });
+  } catch (e) {
+    error(`"${tool}" is not available.`);
+    process.exit(1);
+  }
+}
+ok("required tools installed")
+
+const node_version_out = child_process.execSync(`node -v`);
+const node_version = node_version_out.toString().trim()
+if (node_version < "v20.0") {
+  error(`Want node > 20. Got "${node_version}"`);
+  process.exit(1);
+}
 
 if (!fs.existsSync("vscode")) {
-  child_process.execSync(`git clone --depth 1 https://github.com/microsoft/vscode.git -b ${vscodeVersion}`, {
+  note("cloning vscode")
+  exec(`git clone --depth 1 https://github.com/microsoft/vscode.git -b ${VSCODE_VERSION}`, {
     stdio: "inherit",
   });
+} else {
+  ok("vscodium already installed")
+  note("delete vscodium folder to clone again")
 }
+
+note("changing directory to vscode")
 process.chdir("vscode");
 
 if (!fs.existsSync("node_modules")) {
-  child_process.execSync("yarn", { stdio: "inherit" });
+  exec("yarn", { stdio: "inherit" });
+} else {
+  ok("node_modules exists. Skipping yarn")
 }
+
 // Use simple workbench
-fs.copyFileSync(
-  "../workbench.ts",
-  "src/vs/code/browser/workbench/workbench.ts"
-);
+note("copying workbench file")
+fs.copyFileSync("../workbench.ts", "src/vs/code/browser/workbench/workbench.ts");
 
 // Compile
-child_process.execSync("yarn gulp vscode-web-min", { stdio: "inherit" });
+note("starting compile")
+exec("yarn gulp vscode-web-min", { stdio: "inherit" });
+ok("compile completed")
 
 // Extract compiled files
 if (fs.existsSync("../dist")) {
-  fs.rmdirSync("../dist", { recursive: true });
+  note("cleaning ../dist")
+  fs.rmSync("../dist", { recursive: true, force: true });
+} else {
+  ok("../dist did not exist. No need to clean")
 }
+
 fs.mkdirSync("../dist");
-fse.copySync("../vscode-web", "../dist");
+fs.cpSync("../vscode-web", "../dist", { recursive: true, force: true });
+ok("copied ../vscode-web to ../dist")
+
+// Patch output
+note("beginning complex patches for networking")
+const distWorkbenchPath = "../dist/out/vs/workbench/workbench.web.main.js";
+const workbench = fs.readFileSync(distWorkbenchPath, "utf8");
+const workbenchPatched = workbench.replace(/("https:\/\/{{uuid}}[^"]+")/g, "(globalThis.VSCODE_WEB_PUBLIC_URL||$1)");
+fs.writeFileSync(distWorkbenchPath, workbenchPatched, "utf8");
 
+// Strip sourcemaps and patch vscode-uri for "://" URIs
+const re = /.=function\(\)\{function .\((.),.,(.),.,(.),.\)\{/g;
```

Comment:
I understand the point of removing sourcemaps, but I don't really like these kind of algorithm, hard to maintain.
I'd rather patch the source before compilation to avoid generating these sourcemaps

in `vscode/vs/build/lib/optimize.ts` replacing `sourcemap: 'external'` with `sourcemap: false` should be enough

Reply:
Very cool! Wasn't aware of that optimize logic. Thats for sure way better to handle that upfront. The source maps change doesn't seem to make a big impact overall. When i compared the final size of the dist it was not much of a difference.